### ISSUE TITLE
taint: add @psalm-taint-escape sql for Connection::escape()

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#534]
## Problem

`Illuminate\Database\Connection::escape()` is Laravel's SQL value escaper, using `PDO::quote()` internally. It handles null, int, float, bool, string, and binary types with proper escaping.

The existing `stubs/taintAnalysis/Database/Connection.stubphp` only declares taint **sinks** (query methods) but does not mark `escape()` as a taint **escape** for SQL.

## Current state

`Connection.stubphp` has 13 `@psalm-taint-sink sql` annotations for query methods but zero `@psalm-taint-escape` annotations.

## Proposed fix

Add to `stubs/taintAnalysis/Database/Connection.stubphp`:

```php
/**
 * @psalm-taint-escape sql
 */
public function escape($value, $binary = false): string {}
```

## Methods

- `escape($value, $binary = false): string` — `Illuminate\Database\Connection`

## Impact

Medium — prevents false positives when developers properly escape values before embedding in raw queries.
